Mage.Sets/src/mage/cards/r/RazorfieldRipper.java

```diff
@@ -0,0 +1,139 @@
+        this.addAbility(ability);
+
+        // Reconfigure--Pay {2} or {E}{E}{E}.
+        Cost cost = new OrCost("Pay {2} or {E}{E}{E}", new GenericManaCost(2), new PayEnergyCost(3));
+        this.addAbility(new ReconfigureAbility(cost));
+
+    }
+
+    private RazorfieldRipper(final RazorfieldRipper card) {
+        super(card);
+    }
+
+    @Override
+    public RazorfieldRipper copy() {
+        return new RazorfieldRipper(this);
+    }
+}
+
+class RazorfieldRipperEffect extends OneShotEffect {
+
+    RazorfieldRipperEffect() {
+        super(Outcome.Benefit);
+    }
+
+    private RazorfieldRipperEffect(final RazorfieldRipperEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public RazorfieldRipperEffect copy() {
+        return new RazorfieldRipperEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+
+        game.informPlayers("Target: " + getTargetPointer().toString());
+        int xValue = player.getCountersCount(CounterType.ENERGY);
+        Effect boostEffect  = new BoostTargetEffect(xValue, 3, Duration.EndOfTurn);
+        boostEffect.setTargetPointer(getTargetPointer());
+        boostEffect.apply(game, source);
+        return true;
+    }
+}
+
+class RazorfieldRipperTriggeredAbility extends TriggeredAbilityImpl {
+
+    RazorfieldRipperTriggeredAbility(Effect effect) {
+        super(Zone.BATTLEFIELD, effect);
+        setTriggerPhrase("Whenever {this} or equipped creature attacks");
+    }
+
+    private RazorfieldRipperTriggeredAbility(final RazorfieldRipperTriggeredAbility ability) {
+        super(ability);
+    }
+
+    @Override
+    public RazorfieldRipperTriggeredAbility copy() {
+        return new RazorfieldRipperTriggeredAbility(this);
+    }
+
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.DECLARED_ATTACKERS;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        UUID attacker;
+        if (!game.getCombat().getAttackers().contains(getSourceId())) {
+            Permanent permanent = getSourcePermanentOrLKI(game);
+            if (permanent != null && game.getCombat().getAttackers().contains(permanent.getAttachedTo())) {
+                attacker = permanent.getAttachedTo();
+            } else {
+                attacker = null;
+            }
+        } else {
+            attacker = getSourceId();
+        }
+        game.informPlayers("Effects: " + getEffects().toString());
+        getEffects().setTargetPointer(new FixedTarget(attacker, game));
+        return attacker != null;
+    }
+
+    @Override
+    public String getRule() {
+        return "Whenever {this} or equipped creature attacks, you get {E}, then it gets +X/+X until end of turn, where X is the amount of {E} you have.";
```

Comment:
don't override getRule - just set the text on the boost effect

Reply:
removed